packages/@aws-cdk/aws-vpcv2-alpha/lib/ipam.ts

```diff
@@ -0,0 +1,537 @@
+   * CIDR Mask for Vpc
+   * Only required when using AWS Ipam
+   *
+   * @default - None
+   */
+  readonly netmaskLength?: number;
+
+  /**
+   * Ipv4 or an Ipv6 IPAM pool
+   * Only required when using AWS Ipam
+   *
+   * @default - None
+   */
+  readonly ipamPool?: IIpamPool;
+}
+
+/**
+ * Interface for IpamScope Class
+ */
+export interface IIpamScopeBase{
+
+  /**
+   * Reference to the current scope of stack to be passed in order to create
+   * a new IPAM pool
+   */
+  readonly scope: Construct;
+
+  /**
+   * Default Scope ids created by the IPAM or a new Resource id
+   */
+  readonly scopeId: string;
+
+  /**
+   * Properties to configure IPAMScope
+   */
+  readonly props?: IpamScopeProps;
+
+  /**
+   * Defines scope type can be either default or custom
+   */
+  readonly scopeType?: IpamScopeType;
+
+  /**
+   * Function to add a new pool to an IPAM scope
+   */
+  addPool(id: string, options: PoolOptions): IIpamPool;
+
+}
+
+/**
+ * Creates new IPAM Pool
+ * Pools enable you to organize your IP addresses according to your routing and security needs
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampool.html
+ * @resource AWS::EC2::IPAMPool
+ * @internal
+ */
+class IpamPool extends Resource implements IIpamPool {
+
+  /**
+   * Pool ID to be passed to the VPC construct
+   * @attribute IpamPoolId
+   */
+  public readonly ipamPoolId: string;
+
+  /**
+   * Pool CIDR for IPv6 to be provisioned with Public IP source set to 'Amazon'
+   */
+  public readonly ipamCidrs: CfnIPAMPoolCidr[] = []
+
+  /**
+   * Reference to ipamPool resource created in this class
+   */
+  private readonly _ipamPool: CfnIPAMPool;
+
+  constructor(scope: Construct, id: string, props: IpamPoolProps) {
+    super(scope, id, {
+      physicalName: props.ipamPoolName ?? Lazy.string({
+        produce: () => Names.uniqueResourceName(this, { maxLength: 128, allowedSpecialCharacters: '_' }),
+      }),
+    });
+
+    if (props.addressFamily === AddressFamily.IP_V6 && !props.awsService) {
+      throw new Error('awsService is required when addressFamily is set to ipv6');
+    }
+
+    this._ipamPool = new CfnIPAMPool(this, id, {
+      addressFamily: props.addressFamily,
+      provisionedCidrs: props.provisionedCidrs?.map(cidr => ({ cidr })),
+      locale: props.locale,
+      ipamScopeId: props.ipamScopeId,
+      publicIpSource: props.publicIpSource,
+      awsService: props.awsService,
+    });
+    this.ipamPoolId = this._ipamPool.attrIpamPoolId;
+  }
+
+  /**
+   * A CIDR provisioned to an IPAM pool.
+   * @param id Name of Resource
+   * @param options Either a CIDR or netmask length must be provided
+   * @returns AWS::EC2::IPAMPoolCidr
+   */
+  public provisionCidr(id: string, options: IpamPoolCidrProvisioningOptions): CfnIPAMPoolCidr {
+    const cidr = new CfnIPAMPoolCidr(this, id, {
+      ...options,
+      ipamPoolId: this.ipamPoolId,
+    });
+    this.ipamCidrs.push(cidr);
+    return cidr;
+  }
```

Comment:
This can be used for both ipv4 and ipv6 but mainly used to provision CIDRs with netmasklength as mentioned in the [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-ipampoolcidr.html)
`Can be used for provisioning Amazon-provided IPv6 CIDRs to top-level pools and for provisioning CIDRs to pools with source pools. Cannot be used to provision BYOIP CIDRs to top-level pools. "NetmaskLength" or "Cidr" is required.`

So I believe the validation should be around netmask length, added a point to think about all the validations in IPAM class.